aceps/contact_channels.md

```diff
@@ -0,0 +1,308 @@
+## Contact Channels with delegated tokens
+Date: 2025-05-19
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+<!-- /toc -->
+
+## Notes
+
+I'll refer in a few places to the "humanlayer API", but the integrating party here can be any
+Agent-to-Human (A2H) compatible API, supporting FunctionCall and HumanContact resources.
+
+## Summary
+
+Update the existing server.go endpoints to support delegated tokens via contact channels.
+
+In short, today a client can POST
+
+Example:
+```json
+{
+    "namespace": "default",
+    "agentName": "my-agent",
+    "userMessage": "Hello, how can you help me today?",
+    "responseURL": "https://api.example.com/a2h/v1/human_contacts?response_token=...."
+}
+```
+
+This results in a task being created in the  namespace with 
+
+```
+# ...
+spec:
+  responseURL: https://api.example.com/a2h/v1/human_contacts?response_token=....
+  # ...
+```
+
+And the responseURL is expected to accept a HumanContact resource via POST on the return payload
+
+
+### alternative approach
+
+Instead, accept a `channelToken` in the request body
+
+```json
+{
+    "namespace": "default",
+    "agentName": "my-agent",
+    "userMessage": "Hello, how can you help me today?",
+    "channelToken": "...."
+}
```

Comment:
thinking now... maybe its token and `baseURL` - like https://api.humanlayer.dev/humanlayer/v1 - or https://api.example.com/a2h/v1 - anything that implements 

```
POST $BASE/human_contacts
GET $BASE/human_contacts/{id}
POST $BASE/function_calls
GET $BASE/function_calls/{id}
```